cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+        },
+        'xtriggers': {
+            '__MANY__': [VDR.V_XTRIGGER],
+        },
+        'graph': {
+            '__MANY__': [VDR.V_STRING],
+        },
+    },
+    'runtime': {
+        '__MANY__': {
+            'inherit': [VDR.V_STRING_LIST],
+            'init-script': [VDR.V_STRING],
+            'env-script': [VDR.V_STRING],
+            'err-script': [VDR.V_STRING],
+            'exit-script': [VDR.V_STRING],
+            'pre-script': [VDR.V_STRING],
+            'script': [VDR.V_STRING],
+            'post-script': [VDR.V_STRING],
+            'extra log files': [VDR.V_STRING_LIST],
+            'work sub-directory': [VDR.V_STRING],
+            'meta': {
+                'title': [VDR.V_STRING, ''],
+                'description': [VDR.V_STRING, ''],
+                'URL': [VDR.V_STRING, ''],
+                '__MANY__': [VDR.V_STRING, ''],
+            },
+            'simulation': {
+                'default run length': [VDR.V_INTERVAL, DurationFloat(10)],
+                'speedup factor': [VDR.V_FLOAT],
+                'time limit buffer': [VDR.V_INTERVAL, DurationFloat(30)],
+                'fail cycle points': [VDR.V_STRING_LIST],
+                'fail try 1 only': [VDR.V_BOOLEAN, True],
+                'disable task event handlers': [VDR.V_BOOLEAN, True],
+            },
+            'environment filter': {
+                'include': [VDR.V_STRING_LIST],
+                'exclude': [VDR.V_STRING_LIST],
+            },
+            'job': {
+                'batch system': [VDR.V_STRING, 'background'],
+                'batch submit command template': [VDR.V_STRING],
+                'execution polling intervals': [VDR.V_INTERVAL_LIST, None],
+                'execution retry delays': [VDR.V_INTERVAL_LIST, None],
+                'execution time limit': [VDR.V_INTERVAL],
+                'submission polling intervals': [
+                    VDR.V_INTERVAL_LIST, None
+                ],
+                'submission retry delays': [VDR.V_INTERVAL_LIST, None],
+                'host': [VDR.V_STRING],
+                'owner': [VDR.V_STRING],
+                'platform': [VDR.V_STRING],
+                'suite definition directory': [VDR.V_STRING],
+                'retrieve job logs': [VDR.V_BOOLEAN],
+                'retrieve job logs max size': [VDR.V_STRING],
+                'retrieve job logs retry delays': [
+                    VDR.V_INTERVAL_LIST, None],
+            },
+            'events': {
+                'execution timeout': [VDR.V_INTERVAL],
+                'handlers': [VDR.V_STRING_LIST, None],
+                'handler events': [VDR.V_STRING_LIST, None],
+                'handler retry delays': [VDR.V_INTERVAL_LIST, None],
+                'mail events': [VDR.V_STRING_LIST, None],
+                'mail from': [VDR.V_STRING],
+                'mail retry delays': [VDR.V_INTERVAL_LIST, None],
+                'mail smtp': [VDR.V_STRING],
+                'mail to': [VDR.V_STRING],
```

Comment:
May be out of scope here, but perhaps we need a separate `[mail]` section for the common mail stuffs.

Reply:
It probably __is__ out of scope, but we might as well do it all now. Sounds relatively straightforward.

Reply:
Dun.